src/pyfmi/fmi_util.pyx

```diff
@@ -1089,12 +1088,58 @@ def read_trajectory(file_name, int data_index, int file_position, int sizeof_typ
     cdef unsigned long int start_point = data_index*sizeof_type
     cdef unsigned long int end_point   = sizeof_type*(nbr_points*nbr_variables)
     cdef unsigned long int interval    = sizeof_type*nbr_variables
+
+    if sizeof_type == 4:
+        return _read_trajectory32(file_name, start_point,end_point, interval, file_position, nbr_points)
+    elif sizeof_type == 8:
+        return _read_trajectory64(file_name, start_point,end_point, interval, file_position, nbr_points)
+    else:
+        raise JIOError("Cannot read data that is not of size 32 or 64 bit (float).")
```

Comment:
What about adding something more to the message: "Failed to read the result. The result is on an unsupported format. Can only read data that is either a 32 or 64bit double."